test/e2e/util_test.go

```diff
@@ -472,17 +476,31 @@ func verifyExternalIngressController(t *testing.T, name types.NamespacedName, ho
 		}
 	}()
 
+	// If we have a DNS as an external IP address, make sure we can resolve it before moving on.
+	// This just limits the number of "could not resolve host" errors which can be confusing.
+	if net.ParseIP(address) == nil {
+		if err := wait.PollImmediate(10*time.Second, 5*time.Minute, func() (bool, error) {
+			_, err := net.LookupIP(address)
+			if err != nil {
+				t.Logf("waiting for loadbalancer domain %s to resolve...", address)
+				return false, nil
+			}
+			return true, nil
+		}); err != nil {
+			t.Fatalf("loadbalancer domain %s was unable to resolve:", address)
+		}
+	}
+
 	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("http://%s", address), nil)
 	if err != nil {
 		t.Fatalf("failed to build client request: %v", err)
 	}
-
 	// we use HOST header to map to the domain associated on the ingresscontroller.
 	// This ensures our http call is routed to the correct router.
 	req.Host = hostname
 
 	httpClient := http.Client{Timeout: 5 * time.Second}
-	err = waitForHTTPClientCondition(t, &httpClient, req, 10*time.Second, 5*time.Minute, func(r *http.Response) bool {
+	err = waitForHTTPClientCondition(t, &httpClient, req, 10*time.Second, 10*time.Minute, func(r *http.Response) bool {
```

Comment:
What do we gain with doubling the timeout?  Does just adding 1m help?

Reply:
So yes, 10 mins is pretty aggressive, but I've seen AWS, Azure, GCP take over 5 mins. Here's Miciah's answer to when I asked why it failed at 5 mins: https://github.com/openshift/cluster-ingress-operator/pull/845#issuecomment-1281846948. 

It needs to be at least 7 mins, since we've seen loadbalancers take 6 mins to initialize, but we've been having issues with very long initialization periods for loadbalancers in this test. I could reduce to 7 or 8 mins, but I don't really see the point if it introduces even more flakes. There isn't much to gain by reducing the wait time here, it's parallel. 

I'm hesitant, but I can go down a minute or two if you think there is value.

Reply:
>We might just need a longer timeout on the polling loop; the scope change is known to take as long as ~6 minutes on AWS...

I think we should take a minute to consider changes that add 10-15 minutes to a single test.  We're already waiting up to 3 hrs on some tests due to things we can't control.  

Are customers willing to wait up to 10 minutes for a loadbalancer to initialize?  If not, we need to find out why this activity fails to perform on time, not just give it more time to pass CI.  Just my opinion.

Reply:
> I think we should take a minute to consider changes that add 10-15 minutes to a single test. We're already waiting up to 3 hrs on some tests due to things we can't control.

This only increases the job runtime if (a) this is the longest running test and (b) the polling loop times out.  If the LB is ready after 2 minutes, then it makes no different whether we specify a timeout of 5 minutes or 10 minutes.  If the LB takes 6 minutes to be ready (as is [known to be possible on AWS](https://bugzilla.redhat.com/show_bug.cgi?id=2034795)), then setting the timeout <10 minutes can increase frequency of flakes, and then you get to wait for the whole E2E job to rerun.  

> Are customers willing to wait up to 10 minutes for a loadbalancer to initialize? If not, we need to find out why this activity fails to perform on time, not just give it more time to pass CI. Just my opinion.

If there's a regression, it should be tracked as a separate bug.  As Grant said, optimizing LB initialization is out of scope for this PR.  The goal here is to get CI working again ASAP.